src/python/pants/backend/docker/util_rules/docker_binary.py

```diff
@@ -97,10 +99,14 @@ class DockerBinaryRequest:
 
 
 @rule(desc="Finding the `docker` binary", level=LogLevel.DEBUG)
-async def find_docker(docker_request: DockerBinaryRequest) -> DockerBinary:
+async def find_docker(
+    docker_request: DockerBinaryRequest, docker_options: DockerOptions
+) -> DockerBinary:
+    env = await Get(Environment, EnvironmentRequest(["PATH"]))
+    search_path = docker_options.executable_search_path(env)
     request = BinaryPathRequest(
         binary_name="docker",
-        search_path=docker_request.search_path,
+        search_path=search_path or docker_request.search_path,
```

Comment:
We probably shouldn't search `docker_request.search_path`, anymore, right? It would mean if a user set the paths to `[]`, we would still search some paths. That's nonsensical for a user to do, but also we probably should not do something magical.

Reply:
Hmm, ok. Can address in the related tools pr.